components/PlayerDetailsList/PlayerDetailsList.tsx

```diff
@@ -37,54 +37,50 @@ function PlayerDetailsList(props: Props) {
     let forceUpdate = useForceUpdate()
     let router = useRouter()
 
-    let [listElements, setListElements] = useState<(Auction | BidForList)[]>(props.auctions || [])
-    let [allElementsLoaded, setAllElementsLoaded] = useState(props.auctions ? props.auctions.length < 12 : false)
-    let [playerName, setPlayerName] = useState('')
+    let [listElements, setListElements] = useState<(Auction | BidForList)[]>([])
+    let [allElementsLoaded, setAllElementsLoaded] = useState<boolean>()
+    let [playerName, setPlayerName] = useState<string>('')
     let isLoadingElements = useRef(false)
 
     useEffect(() => {
+
         mounted = true
         router.events.on('routeChangeStart', onRouteChange)
 
+        setPlayerName(props.playerName)
+
         let listState = getListState()
-        if (listState !== undefined) {
+        if (listState) {
             setListElements(listState.listElements)
             setAllElementsLoaded(listState.allElementsLoaded)
-            setTimeout(() => {
-                if (!mounted) {
-                    return
-                }
-                window.scrollTo({
-                    left: 0,
-                    top: listState!.yOffset,
-                    behavior: 'auto'
-                })
-            }, 100)
+            isLoadingElements.current = false
+        }
+        else {
+            // We need to set reset to true to remove the previous player's listElements
+            loadNewElements(true)
```

Comment:
As we preload the auctions again, this could have unwanted site effects.
You change the state here, to trigger a rerender, right?

There is a useful hook (`useForceUpdate`) which can be used here